packages/gatsby-remark-embed-snippet/README.md

```diff
@@ -332,3 +270,119 @@ function App() {
   )
 }
 ```
+
+### Specifying snippets by name
+
+As an alternative to selecting a range of lines from a file, you can add `start-snippet{snippet-name}` and `end-snippet{snippet-name}` in comments in your files. The inclusion of a name for a snippet allows you to create an example file that contains multiple snippets that you reference from different places.
+
+You can specify that you want to only include a named snippet from the embed by using the syntax `{snippet: "snippet-name"}`.
+
+**Javascript example**:
+
+```markdown
+The function to use is:
+
+`embed:api.js{snippet: "funcA"}`
+
+And it is invoked via
+
+`embed:api.js{snippet: "invokeA"}`
+```
+
+With this example file `api.js`:
+
+```javascript
+// start-snippet{funcA}
+function factorial(x) {
+    if (x <= 1) return 1
+    else return x * factorial(x - 1)
+}
+// end-snippet{funcA}
+
+function display() {
+    let x = 5
+    // start-snippet{invokeA}
+    let xfact = factorial(x)
+    // end-snippet{invokeA}
+    println!(`{} factorial is {}`, x, xfact)
+}
+```
+
+Will produce something like this:
+
+```
+The function to use is
```

Comment:
```suggestion
The function to use is:
```

should match the source text...

i miss somehow a `:` at second line `And it is invoked via`